### PR TITLE
wdio-cli: add glob support for the exclude param

### PR DESCRIPTION
**Pull request by schuttsm**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)
On wdio-cli, there's a bug where exclude param is a regex match, not a glob match.  My fix will change it so it attempts a glob match first, if there are no matches, it will attempt a regex match, so as to be backwards compatible.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-config/src/lib/ConfigParser.js [hunk 1]
```diff
@@ -214,8 +214,17 @@ export default class ConfigParser {
         const filesToFilter = new Set()
         const fileList = ConfigParser.getFilePaths(config)
         cliArgFileList.forEach(filteredFile => {
+            let matchedFiles = ConfigParser.getFilePaths(glob.sync(filteredFile))
+            let globMatchedFiles = []
+            matchedFiles.forEach(file => {
+                if (matchedFiles.indexOf(file) != -1) {
```
Comment (baruchvlz) [inline packages/wdio-config/src/lib/ConfigParser.js]: I don't get this statement. You're verifying that the array you're looping has the value that you got from that same array?

File: packages/wdio-config/src/lib/ConfigParser.js [hunk 2]
```diff
@@ -214,8 +214,17 @@ export default class ConfigParser {
...
+                    globMatchedFiles.push(file)
+                }
+            })
```
Comment (baruchvlz) [inline packages/wdio-config/src/lib/ConfigParser.js]: This whole loop is weird. Could you explain the reason behind it?

File: packages/wdio-config/src/lib/ConfigParser.js [hunk 3]
```diff
@@ -214,8 +214,17 @@ export default class ConfigParser {
...
             if (fs.existsSync(filteredFile) && fs.lstatSync(filteredFile).isFile()) {
                 filesToFilter.add(path.resolve(process.cwd(), filteredFile))
+            } else if (globMatchedFiles.length) {
+                globMatchedFiles.forEach(file => {filesToFilter.add(file)})
```
Comment (baruchvlz) [inline packages/wdio-config/src/lib/ConfigParser.js]: You can do something like this if you want to do a one liner:

```js
globMatchedFiles.forEach(filesToFilter.add)
```

File: packages/wdio-config/src/lib/ConfigParser.js (hunk 1 above)
Comment (schuttsm) [reply packages/wdio-config/src/lib/ConfigParser.js]: @baruchvlz - it was left over from a previous refactoring.  You're right, it doesn't make sense.

Let me remove that section & clean it up.

**Comment by schuttsm** [comment]
@baruchvlz thank you for the feedback.  I removed the loop you asked about and I think its ready for another look

**Comment by christian-bromann** [comment]
Thanks @schuttsm for the contribution! 👍

**Pull request merged**